File: ProjetPI(back)/src/main/java/tn/esprit/backend/Entite/Feedback.java
package tn.esprit.backend.Entite;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.*;

import java.util.Date;
import java.util.Set;

@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
@Getter
@Setter
public class Feedback {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private long idFeedback;
    private String name;
    private String surname;
    private String email ;


    @Column(columnDefinition = "TEXT")
    private String acontent;

    @Temporal(TemporalType.TIMESTAMP)
    private Date dateSubmitted;

    private double moyrating;

    @JsonIgnore
    @OneToMany(cascade = CascadeType.ALL)
    private Set<Rating> ratings;
}

File: ProjetPI(front)/src/app/components/models/feedback.ts
export interface Feedback {

    idFeedback: number;
    name: string;
    surname: string;
    email: string;
    acontent: string;
    dateSubmitted: Date;
    ratings?: Rating[];

}


export class Rating {
    idRating!:number;
    idUser!:number;
    idCandidature!:number;
    status!:string;
    moyrating!:number;
}

